Sources/MongoQueue/RecurringTask.swift

```diff
@@ -0,0 +1,93 @@
+import Logging
+import MongoCore
+import Foundation
+import Meow
+
+public protocol RecurringTask: _QueuedTask {
+    /// The moment that you want this to be executed on (delay)
+    /// If you want it to be immediate, use `Date()`
+    var initialTaskExecutionDate: Date { get }
+    
+    /// If you want only one task of this type to exist, use a static task key
+    /// If you want to have many tasks, but not duplicate the task, identify this task by the task key
+    /// If you don't want this task to be uniquely identified, and you want to spawn many of them, use `UUID().uuidString`
+    var uniqueTaskKey: String { get }
+    var taskExecutionDeadline: TimeInterval? { get }
+    
+    func getNextRecurringTaskDate(_ context: ExecutionContext) async throws -> Date?
```

Comment:
@Joannis  I modified this to return a `Date` vs a `TimeInterval` because the next execution date should be determined by the caller and not simply the current date adding some arbitrary time interval. This is helpful for once per day at the same time recurring events where the next interval shouldn't have to calculate itself based on the execution time it took to do its work but should be just a `Date` that the caller determines.

Reply:
That makes sense. We should add an assertion to make sure the new date is ahead of now